gc/base/MemoryPoolBumpPointer.cpp

```diff
@@ -102,11 +102,11 @@ MM_MemoryPoolBumpPointer::internalAllocate(MM_EnvironmentBase *env, uintptr_t si
 		_allocatePointer = (void *)((uintptr_t)_allocatePointer + sizeInBytesRequired);
 		uintptr_t newFreeSpace = spaceRemaining - sizeInBytesRequired;
 		if (0 == newFreeSpace) {
-			_freeEntryCount = 0;
-		} else {
-			_freeEntryCount = 1;
+			_freeEntryCount -= 1;
```

Comment:
Would it be reasonable to add assertions to control that counter is not 0 before decrement?
@amicic What do you think?